BlazarData/src/main/java/com/hubspot/blazar/data/service/BranchStatusService.java

```diff
@@ -0,0 +1,108 @@
+package com.hubspot.blazar.data.service;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.hubspot.blazar.base.GitInfo;
+import com.hubspot.blazar.base.MalformedFile;
+import com.hubspot.blazar.base.ModuleBuild;
+import com.hubspot.blazar.base.ModuleBuildInfo;
+import com.hubspot.blazar.base.ModuleState;
+import com.hubspot.blazar.base.RepositoryBuild;
+import com.hubspot.blazar.base.branch.BranchStatus;
+import com.hubspot.blazar.data.dao.BranchDao;
+import com.hubspot.blazar.data.dao.MalformedFileDao;
+import com.hubspot.blazar.data.dao.RepositoryBuildDao;
+import com.hubspot.blazar.data.dao.StateDao;
+
+public class BranchStatusService {
+  private static final Logger LOG = LoggerFactory.getLogger(BranchStatusService.class);
+
+  private final BranchDao branchDao;
+  private StateDao stateDao;
+  private MalformedFileDao malformedFileDao;
+  private final RepositoryBuildDao repositoryBuildDao;
+
+  @Inject
+  public BranchStatusService(BranchDao branchDao,
+                             StateDao stateDao,
+                             MalformedFileDao malformedFileDao,
+                             RepositoryBuildDao repositoryBuildDao) {
+    this.branchDao = branchDao;
+    this.stateDao = stateDao;
+    this.malformedFileDao = malformedFileDao;
+    this.repositoryBuildDao = repositoryBuildDao;
+  }
+
+
+  public Optional<BranchStatus> getBranchStatusById(int branchId) {
```

Comment:
@gchomatas This is where the magic happens.
A `BranchStatus` comes from many different places, so we query all the relevant places and assemble that data here. For the `ModuleStates` we still have to get the state in 2 separate queries. Changing that will require keeping additional state in the database, which I felt given the size this PR got to was a bit out of scope.